common/http/src/main/java/io/helidon/common/http/MediaType.java

```diff
@@ -268,21 +351,26 @@ private static MediaType create(String type, String subtype,
             throw new IllegalStateException(
                     "A wildcard type cannot be used with a non-wildcard subtype");
         }
-        Map<String, String> builder = new HashMap<>();
+        Map<String, String> normalizedParameters = new HashMap<>();
         for (Map.Entry<String, String> entry : parameters.entrySet()) {
             String attribute = Tokenizer.normalize(TOKEN_MATCHER, entry.getKey());
-            builder.put(attribute, normalizeParameterValue(attribute, entry.getValue()));
+            normalizedParameters.put(attribute, normalizeParameterValue(attribute, entry.getValue()));
         }
 
-        MediaType mediaType = MediaType.builder()
-                .type(normalizedType)
-                .subtype(normalizedSubtype)
-                .parameters(builder)
-                .build();
+        MediaType mediaType = null;
 
         // Return one of the constants if the media type is a known type.
-        //TODO or else get?
-        return Optional.ofNullable(KNOWN_TYPES.get(mediaType)).orElse(mediaType);
+        if (normalizedParameters.isEmpty()) {
+            mediaType = KNOWN_TYPES.get(normalizedType + '/' + normalizedSubtype);
+        }
```

Comment:
You can avoid the `HashMap` creation in probably the vast majority of cases by testing `parameters.isEmpty()` first.  Then, you can move line 363 up as well and simplify.  e.g.:
```
if (parameters.isEmpty()) {
  mediaType = KNOWN_TYPES.get(normalizedType + '/' + normalizedSubtype);
} else {
  Map<String, String> normalizedParameters = new HashMap<>();
  for (Map.Entry<String, String> entry : parameters.entrySet()) {
    String attribute = Tokenizer.normalize(TOKEN_MATCHER, entry.getKey());
    normalizedParameters.put(attribute, normalizeParameterValue(attribute, entry.getValue()));
  }
}
if (mediaType == null) {
  mediaType = MediaType.builder()
    .type(normalizedType)
    .subtype(normalizedSubtype)
    .parameters(normalizedParameters)
    .build();
}
```

Reply:
fixed.